### PR TITLE
fix type definition of ChatPostMessageArguments

### PR DESCRIPTION
**Pull request by kentac55**
###  Summary

Made `text` property of `ChatPostMessageArguments` optional.

close: #1172

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by kentac55** [comment]
Note: 

If we do `postMessage` without specifying either `text` or `blocks` attribute, there will be no problem with the type check, but the API will return `no_text` error.

To prevent this, we should use generic type like [that](https://github.com/slackapi/node-slack-sdk/issues/496#issuecomment-373735256) I think.

**Comment by seratch** [comment]
Thanks a lot for taking the time to make this pull request 🙇 Can you check my comment in the issue? https://github.com/slackapi/node-slack-sdk/issues/1172#issuecomment-779559834

**Comment by kentac55** [comment]
@seratch added warning logic :hugs: 

Can you check this?

File: packages/web-api/src/WebClient.ts
Comment (seratch) [inline packages/web-api/src/WebClient.ts:624]: Can you add other chat.* methods here like I implemented in Python SDK? https://github.com/slackapi/python-slack-sdk/pull/910/files

**Review by seratch — CHANGES_REQUESTED** [review]
Thanks for updating! Please check my comment.

File: packages/web-api/src/WebClient.ts
Comment (kentac55) [reply packages/web-api/src/WebClient.ts]: Thank you for revieing!

added warning logic for other methods and updated test :smile:

**Comment by kentac55** [comment]
@seratch Sorry for late response. I forgot to mention you 🙇 

Can you check the new changes?

File: packages/web-api/src/WebClient.ts
Comment (seratch) [inline packages/web-api/src/WebClient.ts:627]: Sorry, the implementation I mentioned had lack of consideration for `attachments`. Can you update this logic to check `fallback` in the case of `attachments`? Refer to https://github.com/slackapi/python-slack-sdk/pull/966 for the logic.

**Review by seratch — CHANGES_REQUESTED** [review]

**Comment by seratch** [comment]
@kentac55 Thanks for being patient here and updating the code for the attachments pattern! If you don't mind, could you rename the local variables like this? https://github.com/seratch/node-slack-sdk/commit/b963985c52c0598f2091667620b76d010255e5cc

**Comment by kentac55** [comment]
@seratch Sorry for late response :bow: 

cherry-picked your commit :smiley:

**Review by seratch — APPROVED** [review]
Thanks for working on this! The changes look great to me but I would like to have other maintainers' eyes before merging it.

**Comment by seratch** [comment]
OK, there is no concerns about this. Let me merge this PR. Thanks for your contribution! @kentac55

**Pull request merged**
